src/system/System.ts

```diff
@@ -528,8 +569,18 @@ export default class System {
       return this.settlementRates.get(key)!;
     }
 
-    const settlementRate = await this.notionalProxy.getSettlementRate(currencyId, maturity);
-    if (settlementRate.underlyingDecimals.isZero()) {
+    const settlementRateResponse = await this.graphClient.queryOrThrow<SettlementRateQueryResponse>(
+      settlementRateQuery,
+      {
+        variables: {currencyId, maturity},
+      },
+    );
+    if (settlementRateResponse.settlementRates.length === 0) {
```

Comment:
In the case when settlement rates are not found, we should continue the function to line 584 rather than throw an error. This will legitimately occur in some edge cases. So maybe something like:

```
const isSettlementRateSet = settlementRateResponse.settlementRates.length > 0 && settlementRateResponse.settlementRates[0].assetExchangeRate

if (!isSettlementRateSet) {
   const assetRate = ...
}
````

Reply:
Sounds good, made the change